app/screens/settings-screen/account-limits-screen.tsx

```diff
@@ -111,136 +100,77 @@ export const AccountLimitsScreen = () => {
-                      </Text>
-                      <Text adjustsFontSizeToFit style={styles.valueTotal}>
-                        {`${usdAmountDisplay(
-                          Number(convertLimits?.WeeklyAccountLimit.totalLimit),
-                          0,
-                        )} ${LL.AccountLimitsScreen.perWeek()}`}
-                      </Text>
-                    </View>
-                  </View>
-                ) : null}
-              </View>
-            )
+    <Screen style={styles.container}>
+      <View style={styles.container}>
+        <View style={styles.limitWrapper}>
+          <Text adjustsFontSizeToFit style={styles.valueFieldType}>
+            {LL.AccountLimitsScreen.receive()}
+          </Text>
+          <View style={styles.content}>
+            <View style={styles.contentTextBox}>
+              <Text adjustsFontSizeToFit style={styles.valueRemaining}>
+                {LL.AccountLimitsScreen.unlimited()}
+              </Text>
+            </View>
+          </View>
+        </View>
+
+        <View style={styles.divider}></View>
+
+        <View style={styles.limitWrapper}>
+          <Text adjustsFontSizeToFit style={styles.valueFieldType}>
+            {LL.AccountLimitsScreen.withdraw()}
+          </Text>
+          {limits?.withdrawal.map((data, index: number) => {
+            return <AccountLimitsPeriod data={data} key={index} />
           })}
         </View>
-      </Screen>
-    </ScrollView>
+
+        <View style={styles.divider}></View>
+
+        <View style={styles.limitWrapper}>
+          <Text adjustsFontSizeToFit style={styles.valueFieldType}>
+            {LL.AccountLimitsScreen.internalSend()}
+          </Text>
+          {limits?.internalSend.map((data, index: number) => {
+            return <AccountLimitsPeriod data={data} key={index} />
+          })}
+        </View>
+
+        <View style={styles.divider}></View>
+
+        <View style={styles.limitWrapper}>
+          <Text adjustsFontSizeToFit style={styles.valueFieldType}>
+            {LL.AccountLimitsScreen.stablesatTransfers()}
+          </Text>
+          {limits?.convert.map((data, index: number) => {
+            return <AccountLimitsPeriod data={data} key={index} />
+          })}
+        </View>
+      </View>
+    </Screen>
+  )
+}
+
+const AccountLimitsPeriod = ({ data }: accountLimitsPeriodProps) => {
+  const { LL } = useI18nContext()
+  return (
+    <View style={styles.content}>
+      <View style={styles.contentTextBox}>
+        <Text adjustsFontSizeToFit style={styles.valueRemaining}>
+          {`${usdAmountDisplay(
+            Number(data.remainingLimit),
+            0,
+          )} ${LL.AccountLimitsScreen.remaining().toLocaleLowerCase()}`}
+        </Text>
+        <Text adjustsFontSizeToFit style={styles.valueTotal}>
+          {`${usdAmountDisplay(Number(data.totalLimit), 0)} ${
+            data.__typename === "DailyAccountLimit"
```

Comment:
This will also need to handle monthly limit, so it may be best to use a map or function that maps typename to a string representing the duration. As it is written currently, if a new month limit was added, it would display as a week limit.